src/smc-webapp/markdown.ts

```diff
@@ -39,7 +41,15 @@ See https://github.com/sagemathinc/cocalc/issues/2863
 for another example where remove_math is annoying.
 */
 
-export function markdown_to_html(markdown_string: string): string {
+export interface MD2html {
+  html: string;
+  frontmatter: string;
+}
+
+function process(
+  markdown_string: string,
+  mode: "default" | "frontmatter"
```

Comment:
`process` is a little enhancement of the markdown processing we already have, where the `mode` switches to the one with frontmatter processing. I didn't want to do it unconditionally, because it needs to  instantiate the parser again and  maybe there is some overhead. tl;dr, all existing markdown processing calls behave exactly the same as before.